app/para.py

```
@@ -0,0 +1,216 @@
+
+  for entry in entries:
+    if entry["entryid"] == entryid:
+      return jsonify(entry)
+
+  raise ResponseWithError(
+    'Cannot find entry %s/%s' % (threadid, entryid),
+    status_code=404
+  )
+
+@app.route('/paragraph/multi/<threadids>', methods=['GET'])
+def domultiget(threadids):
+  results = []
+  for threadid in threadids.split(','):
+    record, rev = getimage(threadid)
+    results.append(record)
+
+  return Response(json.dumps(results),  mimetype='application/json')
+
+@app.route('/paragraph/create', methods=['POST'])
+def dopost():
+  req = json.loads(request.get_data())
+
+  # check required fields in create request
+  checkrequired(req, ['threadid','content', 'author', 'passhash'])
+
+  threadid = req['threadid']
+  content  = req['content']
+  author   = req['author']
+  passhash = req['passhash']
+
+  record, rev = getimage(threadid)
+
+  now = int(time.time())
+  entry = {
+    "entryid"  : uuid.uuid4().urn,
+    "modified" : now,
+    "created"  : now,
+    "content"  : content,
+    "author"   : author,
+    "passhash" : passhash
+  }
+
+  # append new entry and write back
+  record['entries'].append(entry)
+  putimage(record, rev)
+
+  # return post-image as result of create
+  return jsonify(entry)
+
+@app.route('/paragraph/<threadid>/<entryid>', methods=['PUT'])
+def doput(threadid, entryid):
+  req = json.loads(request.get_data())
+  req["modified"] = int(time.time())
+
+  checkrequired(req, ['entryid', 'created', 'modified', 'content','author','passhash'])
+
+  record, rev = getimage(threadid)
+
+  entries = record["entries"]
+  target = None
+  for entry in entries:
+    if entry["entryid"] == entryid:
+      target = entry
+
+  # remove existing entry, if password matches
+  if target != None:
+    if target['passhash'] != req['passhash']:
+      raise ResponseWithError(
+        "Password hash does not match, "
+        "cannot proceed with update for %s" % entryid,
+        status_code=401
+      )
+    entries.remove(target)
+
```

the ``doput`` method allows users to append to entries an entry with an entryid/created/modified/content/author/passhash of their own choosing if the request's entryid does not exist in the thread's entries. If ``target == None``, it should throw an exception here.